### PR TITLE
Add `#![deny(missing_docs)]`

### LINKED ISSUE #94: Add #![deny(missing_docs)]
It's nice to have at least some minimal kind of docs for important types and functions.

### PR DESCRIPTION
**Pull request by fitzgen**
This commit adds the `#![deny(missing_docs)]` pragma, which causes
compilation to fail if a public type or function is missing a
documentation comment. It also adds missing documentation comments for
public types and functions that were missing them.

Fixes #94 

r? @emilio

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -45,107 +58,189 @@ use ir::item::{Item, ItemId};
...
+    /// Whitelisted functions. See docs for `whitelisted_types` for more.
     pub whitelisted_functions: RegexSet,
+
+    /// Whitelisted variables. See docs for `whitelisted_types` for more.
     pub whitelisted_vars: RegexSet,
+
+    /// Whether we should generate builtins or not.
     pub builtins: bool,
+
+    /// The set of libraries we should link in the generated Rust code.
```
Comment (emilio) [inline src/lib.rs]: Could you add a FIXME here pointing to #104?

Comment (fitzgen) [inline src/lib.rs]: Done.

**Pull request merged**